volumina/__main__.py

```diff
@@ -0,0 +1,80 @@
+import argparse
+import contextlib
+import signal
+import sys
+
+import numpy
+import xarray
+from PyQt5.QtWidgets import QApplication
+
+from volumina import __version__
+from volumina.api import Viewer
+from volumina.colortables import default16_new
+from volumina.pixelpipeline.datasources import ArraySinkSource, ArraySource
+from volumina.layer import ColortableLayer, GrayscaleLayer
+
+
+@contextlib.contextmanager
+def volumina_viewer():
+    app = QApplication(sys.argv)
+    v = Viewer()
+    v.editor.setInteractionMode("navigation")
+    yield v
+    try:
+        signal.signal(signal.SIGINT, lambda *_: app.quit())
+        app.exec_()
+    finally:
+        app.quit()
+
+
+def axiorder_type(value):
+    reasons = []
+    if len(set(value)) != len(value):
+        reasons.append("Each axis may only appear ones")
+
+    if any(dim not in "txyzc" for dim in value):
+        reasons.append("Unrecognized axis value encountered. Only allowed axes: 'txyzc'.")
+
+    if reasons:
+        reasons.append(f"Got '{value}'.")
+        raise argparse.ArgumentTypeError(" ".join(reasons))
+
+    return value
+
+
+def parse_args():
+    p = argparse.ArgumentParser(
+        description="",
+        usage="",
+        epilog="",
+    )
+    p.add_argument("image", help="Path to .npy image")
+    p.add_argument("axistags", help="Strings describing axes in image. Valid values: 'tzyxc'", type=axiorder_type)
+    p.add_argument("--version", action="version", version=__version__)
+
+    args = p.parse_args()
+    return args
+
+
+def reorder_to_volumina(data, axistags):
+    tagged_data = xarray.DataArray(data, dims=tuple(axistags))
```

Comment:
Are we depending on `xarray` just for axis reordering?

If this is the case, then it's trivial to do this ourselves, and we can drop the dependency.

```python
def normalize_axes(a, axes, *, order=('t', 'x', 'y', 'z', 'c')):
    transpose_axes = [axes.index(d) for d in order if d in axes]
    expand_axes = [i for i, d in enumerate(order) if d not in axes]
    return numpy.expand_dims(numpy.transpose(a, transpose_axes), expand_axes)
```

Reply:
I'd like to keep it (at some point adding an xarray source). The only dependency that might be cumbersome might be pandas that is brought in via xarray.
What reservation towards xarray do you have (we depend on it in ilastik already)?

Reply:
I like xarray, but I'm against adding dependencies for doing trivial things. However, in this case it's probably OK if we are planning to use it more.

Reply:
I want to strongly endorse using it, so yep, I'd like to keep it in here.